openapi/resource-server.yaml

```diff
@@ -1365,5 +1381,21 @@ components:
         example: 'Signature-Input: sig1=("@method" "@target-uri" "content-digest" "content-length" "content-type");created=1618884473;keyid="gnap-rsa"'
       description: 'The Signature-Input field is a Dictionary structured field containing the metadata for one or more message signatures generated from components within the HTTP message.  Each member describes a single message signature.  The member''s key is the label that uniquely identifies the message signature within the context of the HTTP message.  The member''s value is the serialization of the covered components Inner List plus all signature metadata parameters identified by the label.  The following components MUST be included: - "@method" - "@target-uri" - "authorization".  When the message contains a request body, the covered components MUST also include the following: - "content-digest"  The keyid parameter of the signature MUST be set to the kid value of the JWK.      See [ietf-httpbis-message-signatures](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures#section-4.1) for more details.'
       required: true
+    optional-signature:
```

Comment:
This is a nitpick that I don't feel the need to block over, but I wonder if the optional signature parameters could be reduced to a reference to the original required version but with an overridden `required` tag so there's less repeated text.

Something else to note is that Swagger defaults `required` to `false`, so perhaps the optional version should be the 'base' parameter and the required ones should extend those.

Reply:
Good suggestion. I tried doing it like this:

```yaml
    signature-input:
      $ref: '#/components/parameters/optional-signature-input'
      required: true
```

Which seemed to work when running the code, except when generating the types it lost the non-schema details of the reference (so there was no comment with the description, and the key was `undefined` instead of `Signature-Input`, so I had to settle for just re-using the schema and duplicating the rest.